### ISSUE TITLE
addSelect with names?

### ISSUE THREAD

**Issue by darkWolf-PR** [contributte/forms-bootstrap#16]
While doing some forms, I´ve needed one select filled with names instead of keys, with base Nette Forms, this works:
$form->addSelect('country', 'Země:')
	->setItems($countries, false);

When using new BootstrapForm(), it gives me type error:

Argument 1 passed to Contributte\FormsBootstrap\Inputs\SelectInput::setItems() must be of the type array, null given, called in ...\vendor\contributte\forms-bootstrap\src\Inputs\SelectInput.php on line 27

Am I using it wrong, or is it a bug?

**Comment by mskocik** [comment]
Probably a bug but you should be able to workarount it by passing your values directly in constructor like this:
```php
$form->addSelect('country', 'Země', array_values($countries));
```

**Comment by darkWolf-PR** [comment]
Nope, doesn´t work. With array like ['one', 'two',..], options still ends with value=0, value=1 etc.

**Comment by mskocik** [comment]
Actually that's correct behaviour. Because keys are **always** used as values for selects, so the second argument of `setItems` method has meaning **only for associative** arrays.

So if you want use _real_ values as keys you have to do this:

```php
$list= array_combine($list, $list);

// where $list now looks like this:
$list = [
 'one'   => 'one',
 'two'   => 'two',
 'three' => 'three',
];
```
So your _problem_ is not really related to this package from my point of view.

**Comment by darkWolf-PR** [comment]
You are wrong about associative arrays, what I wrote - array('one', 'two',...) used with setItems and second argument on base Nette forms gives me select where names are used as keys, Just like what is stated in Nette Forms docu. Don´t have to pass associative array for that.

> Prvky je rovněž možné přidat pomocí metody setItems(). Pokud chceme místo klíčů položek získat přímo jejich hodnoty, můžeme toho docílit druhým argumentem:

$form->addSelect('country', 'Země:')
	->setItems($countries, false);


If it´s not supposed to work as base Nette Forms(which is strange considering it´s Bootstrap Renderer for Nette Forms), it should be stated in documentation. Otherwise, people waste hours trying something thats actually not supported.

**Comment by dakorpar** [comment]
Looks like a bug in forms-bootstrap, thanks for reporting.
I believe https://github.com/contributte/forms-bootstrap/commit/7de10ce9d00a3e89e6fdeec8092e90cf03b23371 fixes this...
can U try it again?

**Comment by darkWolf-PR** [comment]
@dakorpar Tested, partially works.

I´ve tested on this array:
```php
$countries = [
        'Belgium', 'Australia', 'Austria', 'Belgium',
        'Czech Republic' => [
            'Praha',
            'Brno',
        ],
        'Denmark',
    ];
```

Works for normal array, fails with this two-dimensional one. Sub-array gets removed. Also, if u happen to switch for keys, it crashes with:

"Nette\InvalidArgumentException
Value "0" is used multiple times."

While normal Nette Forms somehow not. Even thought it also has 0 used multiple times there.

**Comment by darkWolf-PR** [comment]
I found another problem, probably somewhat connected, with filling checkboxLists. Not sure if it needs another issue. Works normally in base Nette Forms.

Filling addCheckboxList with this:

        return $this->db->table(self::TABLE_NAME)
                        ->select(('YEAR(date_added) AS year'))
                        ->group('year')
                        ->order('year DESC')
                        ->fetchPairs('year', 'year');

Leads into:

TypeError
Argument 3 passed to Contributte\FormsBootstrap\Inputs\CheckboxInput::makeCheckbox() must be of the type string or null, integer given, called in ...vendor\contributte\forms-bootstrap\src\Inputs\CheckboxListInput.php on line 43

**Comment by dakorpar** [comment]
@darkWolf-PR 
regarding:
```php
$countries = [
        'Belgium', 'Australia', 'Austria', 'Belgium',
        'Czech Republic' => [
            'Praha',
            'Brno',
        ],
        'Denmark',
    ];
```
this is actually feature which saves you, since in the end in nette/forms this ends up with same values for Belgium and Praha (option value="0" for both) or Australia and Brno (option value="1" for both) and I'm pretty sure no one wants this behaviour.

Regarding:

Filling addCheckboxList with this:
```php
        return $this->db->table(self::TABLE_NAME)
                        ->select(('YEAR(date_added) AS year'))
                        ->group('year')
                        ->order('year DESC')
                        ->fetchPairs('year', 'year');
```
is solved in PR #22 and will be released today in v0.2.2,
sorry for slow response I was quite buisy.
